File: src/main/java/InputScannerView.java
import java.util.Scanner;
import java.util.Set;

public class InputScannerView {

    private Scanner scanner;
    private InputScanner inputScanner;

    public InputScannerView(InvertedIndex index) {
        this.scanner = new Scanner(System.in);
        this.inputScanner = new InputScanner(index);
    }

    public void start(String string) {
        if (!string.isEmpty()) {
            while (true) {
                string = scanner.nextLine();
                if (string.equals("--back")) break;
                showResult(inputScanner.getOrder(string));
            }
        }
        scanner.close();
    }

    private void showResult(Set<String> answer) {
        for (String s : answer) System.out.println(s);
    }

}

File: ConsoleApp1/InputScannerView.cs
﻿using System;
using System.Collections.Generic;

namespace ConsoleApp1
{
    public class InputScannerView
    {
        private const string BackString = "--back";

        public InputScannerView(IInvertedIndex index)
        {

            var inputScanner = new InputScanner(index);
            while (true)
            {
                var input = Console.ReadLine();
                if (string.Equals(input, BackString)) break;
                ShowResult(inputScanner.GetOrder(input));
            }
        }

        private static void ShowResult(IEnumerable<string> answer)
        {
            foreach (var s in answer)
            {
                Console.Out.WriteLine(s);
            }
        }
    }
}